File: tests/uasm/sample/realistic/RealisticTest.cs
using UdonSharp;
using UnityEngine;

[UdonBehaviourSyncMode(BehaviourSyncMode.None)]
public class RealisticTest : UdonSharpBehaviour
{
    private int score;
    private int highScore;
    private bool isPlaying;

    public void Start()
    {
        score = 0;
        highScore = 100;
        isPlaying = true;
    }

    public void AddScore(int points)
    {
        if (!isPlaying) return;

        score = score + points;
        if (score > highScore)
        {
            highScore = score;
            Debug.Log("New high score!");
        }

        string msg = "Score: " + score.ToString();
        Debug.Log(msg);
    }

    public void ResetGame()
    {
        score = 0;
        isPlaying = true;
        Debug.Log("Game reset");
    }

    public int GetScore()
    {
        return score;
    }

    public int GetHighScore()
    {
        return highScore;
    }
}

File: tests/uasm/sample/realistic/RealisticTest.ts
import { UdonBehaviour } from "@ootr/udon-assembly-ts/stubs/UdonDecorators";
import { UdonSharpBehaviour } from "@ootr/udon-assembly-ts/stubs/UdonSharpBehaviour";
import type { UdonInt } from "@ootr/udon-assembly-ts/stubs/UdonTypes";
import { Debug } from "@ootr/udon-assembly-ts/stubs/UnityTypes";

@UdonBehaviour()
export class RealisticTest extends UdonSharpBehaviour {
  private score: UdonInt = 0 as UdonInt;
  private highScore: UdonInt = 0 as UdonInt;
  private isPlaying: boolean = false;

  Start(): void {
    this.score = 0 as UdonInt;
    this.highScore = 100 as UdonInt;
    this.isPlaying = true;
  }

  AddScore(points: UdonInt): void {
    if (!this.isPlaying) return;

    this.score = (this.score + points) as UdonInt;
    if (this.score > this.highScore) {
      this.highScore = this.score;
      Debug.Log("New high score!");
    }

    const msg: string = `Score: ${this.score.toString()}`;
    Debug.Log(msg);
  }

  ResetGame(): void {
    this.score = 0 as UdonInt;
    this.isPlaying = true;
    Debug.Log("Game reset");
  }

  GetScore(): UdonInt {
    return this.score;
  }

  GetHighScore(): UdonInt {
    return this.highScore;
  }
}
